src/Support/SupportData.php

```diff
@@ -0,0 +1,930 @@
+			'amp_supported_post_types'    => ( ! empty( $amp_settings['supported_post_types'] ) && is_array( $amp_settings['supported_post_types'] ) ) ? $amp_settings['supported_post_types'] : [],
+			'amp_supported_templates'     => ( ! empty( $amp_settings['supported_templates'] ) && is_array( $amp_settings['supported_templates'] ) ) ? $amp_settings['supported_templates'] : [],
+			'amp_mobile_redirect'         => ( ! empty( $amp_settings['mobile_redirect'] ) ) ? true : false,
+			'amp_reader_theme'            => ( ! empty( $amp_settings['reader_theme'] ) ) ? $amp_settings['reader_theme'] : '',
+		];
+
+		return $site_info;
+	}
+
+	/**
+	 * To get list of active plugin's information.
+	 *
+	 * @since 2.2
+	 *
+	 * @return array List of plugin detail.
+	 */
+	public function get_plugin_info() {
+
+		$active_plugins = get_option( 'active_plugins' );
+
+		if ( is_multisite() ) {
+			$network_wide_activate_plugins = get_site_option( 'active_sitewide_plugins' );
+			$active_plugins                = array_merge( $active_plugins, $network_wide_activate_plugins );
+		}
+
+		$active_plugins = array_values( array_unique( $active_plugins ) );
+		$plugin_info    = array_map( __CLASS__ . '::normalize_plugin_info', $active_plugins );
+		$plugin_info    = array_filter( $plugin_info );
+
+		return array_values( $plugin_info );
+	}
+
+	/**
+	 * To get active theme info.
+	 *
+	 * @since 2.2
+	 *
+	 * @return array List of theme information.
+	 */
+	public function get_theme_info() {
+
+		$themes   = [ wp_get_theme() ];
+		$response = array_map( __CLASS__ . '::normalize_theme_info', $themes );
+		$response = array_filter( $response );
+
+		return array_values( $response );
+	}
+
+	/**
+	 * To get error log.
+	 *
+	 * @since 2.2
+	 *
+	 * @return array Error log contents and log_errors ini setting.
+	 */
+	public function get_error_log() {
+
+		$error_log_path = ini_get( 'error_log' );
+
+		// $error_log_path might be a relative path/filename.
+		// In this case, we would have to iterate many directories to find them.
+		if ( empty( $error_log_path ) || ! file_exists( $error_log_path ) ) {
+			return [
+				'log_errors' => ini_get( 'log_errors' ),
+				'contents'   => '',
+			];
+		}
+
+		$file        = file( $error_log_path );
+		$max_lines   = max( 0, count( $file ) - 200 );
+		$file_length = count( $file );
+		$contents    = [];
+
+		for ( $i = $max_lines; $i < $file_length; $i ++ ) {
+			if ( ! empty( $file[ $i ] ) ) {
+				$contents[] = sanitize_text_field( $file[ $i ] );
+			}
+		}
+
+		return [
+			'log_errors' => ini_get( 'log_errors' ),
+			'contents'   => implode( "\n", $contents ),
+		];
+	}
+
+	/**
+	 * To get plugin information by plugin file.
+	 *
+	 * @since 2.2
+	 *
+	 * @param string $plugin_file Plugin file.
+	 *
+	 * @return array Plugin detail.
+	 */
+	public static function normalize_plugin_info( $plugin_file ) {
+
+		$absolute_plugin_file = WP_PLUGIN_DIR . DIRECTORY_SEPARATOR . $plugin_file;
```

Comment:
Here `DIRECTORY_SEPARATOR` is used, but below it `explode`'s on `'/'`. I think this can be then just:

```suggestion
		$absolute_plugin_file = WP_PLUGIN_DIR . '/' . $plugin_file;
```